### PR TITLE
chore: add migrate and format scripts

### PR DESCRIPTION
**Pull request by zaosoula**

### CODE REVIEW

**Review by julienqueffelec — APPROVED** [review]

File: .husky/pre-commit [hunk 1]
```diff
@@ -2,3 +2,5 @@
 . "$(dirname -- "$0")/_/husky.sh"
 
 yarn format
+yarn migrate
+git add langs/*
```
Comment (kyazdani42) [inline .husky/pre-commit:6]: Not sure about this, pre-committing all changes in a folder is a bit strange to me.

File: bin/commands/format.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+const langFolder = path.join(process.cwd(), 'langs');
+const replacements: ReplacementDict = {
+  spaceBeforeLineBreak: [/ +\n/, '\n'],
+  spaceInPlaceholder: [/\{\{ +(\w+) +\}\}/, '{{$1}}'],
+};
+let changedKeys = 0;
+
+const formatKey = (key: string, value: string) => {
+  let formated = value;
```
Comment (kyazdani42) [inline bin/commands/format.ts:16]: formatted takes 2 `t`

File: bin/commands/format.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    changedKeys += 1;
+  }
+
+  return [key, formated];
+};
+
+const formatDict = (dict: Dict): Dict =>
+  Object.fromEntries(
+    Object.entries(dict).map(([key, value]) => {
+      if (typeof value === 'object') {
```
Comment (kyazdani42) [inline bin/commands/format.ts:32]: will fail for arrays i believe.
`typeof [] === 'object'`
check `Array.isArray` also

File: bin/commands/format.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      return formatKey(key, value);
+    })
+  );
+const sortDict = (dict: Dict): Dict =>
+  Object.keys(dict)
+    .sort()
+    .reduce(
+      (acc, key) => ({
+        ...acc,
+        [key]: typeof dict[key] === 'object' ? sortDict(dict[key] as Dict) : dict[key],
```
Comment (kyazdani42) [inline bin/commands/format.ts:45]: bis

Comment (kyazdani42) [reply bin/commands/format.ts:45]: Although ignore these comments if you cannot have an array in your json object.

**Review by kyazdani42 — APPROVED** [review]

File: .husky/pre-commit (hunk 1 above)
Comment (zaosoula) [reply .husky/pre-commit:6]: as the script are supposed to format, migrate, sort the lang file adding this command includes the changes in the commit

File: bin/commands/format.ts (hunk 3 above)
Comment (zaosoula) [reply bin/commands/format.ts:32]: A

**Pull request merged**